sparse/coo/core.py

```diff
@@ -1465,6 +1515,76 @@ def nonzero(self):
         """
         return tuple(self.coords)
 
+    def asformat(self, format):
+        """
+        Convert this sparse array to a given format.
+
+        Parameters
+        ----------
+        format : str
+            A format string.
+
+        Returns
+        -------
+        out : SparseArray
+            The converted array.
+
+        Raises
+        ------
+        NotImplementedError
+            If the format isn't supported.
+        """
+        if format == 'coo' or format is COO:
+            return self
+
+        from ..dok import DOK
+        if format == 'dok' or format is DOK:
+            return DOK.from_coo(self)
+
+        raise NotImplementedError('The given format is not supported.')
+
+
+def as_coo(x, shape=None):
+    """
+    Converts any given format to :obj:`COO`. See the "See Also" section for details.
+
+    Parameters
+    ----------
+    x : SparseArray or numpy.ndarray or scipy.sparse.spmatrix or Iterable.
+        The item to convert.
+    shape : tuple[int], optional
+        The shape of the output array. Can only be used in case of Iterable.
+
+    Returns
+    -------
+    out : COO
+        The converted :obj:`COO` array.
+
+    See Also
+    --------
+    SparseArray.asformat : A utility function to convert between formats in this library.
+    COO.from_numpy : Convert a Numpy array to :obj:`COO`.
+    COO.from_scipy_sparse : Convert a SciPy sparse matrix to :obj:`COO`.
+    COO.from_iter : Convert an iterable to :obj:`COO`.
+    """
+    if hasattr(x, 'shape') and shape is not None:
+        raise ValueError('Cannot provide a shape in combination with something '
+                         'that already has a shape.')
+
+    if isinstance(x, SparseArray):
+        return x.asformat('coo')
+
+    if isinstance(x, np.ndarray):
+        return COO.from_numpy(x)
+
+    if isinstance(x, scipy.sparse.spmatrix):
+        return COO.from_scipy_sparse(x)
+
+    if isinstance(x, Iterable):
+        return COO.from_iter(x, shape=shape)
+
+    raise NotImplementedError('Format not supported for conversion.')
```

Comment:
It might be helpful to list here what was provided from the user and a set of valid values.

Reply:
Done!